sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusExtensionConfigProvider.cs

```diff
@@ -108,6 +110,23 @@ public void Initialize(ExtensionConfigContext context)
             context.AddBindingRule<ServiceBusAttribute>().Bind(bindingProvider);
         }
 
+        internal static ParameterBindingData ConvertReceivedMessageToBindingData(ServiceBusReceivedMessage message)
+        {
+            ReadOnlyMemory<byte> messageBytes = message.ToAmqpBytes().ToMemory();
+            byte[] lockTokenBytes = Guid.Parse(message.LockToken).ToByteArray();
+            const int lockTokenLength = 16;
+
+            byte[] combinedBytes = new byte[messageBytes.Length + lockTokenLength];
+
+            // The 16 lock token bytes go in the beginning
+            lockTokenBytes.CopyTo(combinedBytes.AsSpan());
+
+            // The AMQP message bytes go after the lock token bytes
+            messageBytes.CopyTo(combinedBytes.AsMemory(lockTokenLength));
+
+            return new ParameterBindingData("1.0", "AzureServiceBus", BinaryData.FromBytes(combinedBytes), "application/octet-stream");
```

Comment:
I was following the pattern from https://github.com/Azure/azure-sdk-for-net/pull/31757/files#diff-e323d68130e56f27c5b978175439cd66ade4dba43a7bcaf0e64ffcf89d03b8a4R9 where the extension name is used. I believe the consumer code would only get this particular form of ParameterBindingData, so the source is really just metadata. If a schema change were possible, we'd probably key off of the version.

Reply:
That said, I don't think it would hurt to be more specific and it could potentially be helpful in the future to disambiguate. @liliankasem do you have thoughts on this?

Reply:
Currently, source is only used to determine which converter we need to use to handle the content provided (i.e. to make sure we use the service bus specific converter vs the blob or string converters) so based on that I don't think we need to be more specific. I would expect the content to have any information we need about the message type? 

If we were more specific, i.e. `AzureServiceBusMessage` - what else would be something that you would potentially send over via ParameterBindingData? Just trying to think of scenarios where we would ever need the source to disambiguate

Reply:
Based on the `[Extension("ServiceBus")]` header used here, I would expect this value to just be `ServiceBus`

Reply:
We will need a converter for going in the other direction for the output case and this converter acts on a different message type - ServiceBusMessage. The Content contains binary data - so it would be better to know up front what we are working on rather than attempting to parse it to determine.

Reply:
I think with the current design we would have one converter for all scenarios (input/output/trigger), so source is only there to tell us that it is service bus, but you make an interesting point. We don't have plans for output binding support at the moment and we are hoping to encourage customers to use the SDK type they already have to handle any output rather than relying on output bindings - but that's not to say we won't have output binding support, just out of scope for now so we just haven't spent much time considering the output binding design really